test/e2e/performance_test.go

```diff
@@ -0,0 +1,365 @@
+// `--http.performance.concurrency` concurrency to the Nginx Pod. `workloads` indicates how many CIDRs
+// in the workload network policy should be generated.
+func httpRequest(times int, workloads int, data *TestData, b *testing.B) {
+	nginxPodIP, _ := setupPerformanceTestPods(data, b)
+	// enable Pods connectivity policy first
+	if err := setupPerformanceTestPodsConnection(data); err != nil {
+		b.Fatalf("Error when adding network policy to set up connection between performance test Pods")
+	}
+	b.Log("Populating performance test workloads")
+	if err := populateWorkload(generateWorkload(workloads), data); err != nil {
+		b.Fatalf("Error when populating workloads: %v", err)
+	}
+
+	if err := waitNetworkPolicyRealize(workloads, data); err != nil {
+		b.Fatalf("Checking network policies realization failed: %v", err)
+	} else {
+		b.Log("Network policy realized")
+	}
+
+	serverURL := &url.URL{Scheme: "http", Host: nginxPodIP, Path: "/"}
+	b.StartTimer()
+	for i := 0; i < b.N; i++ {
+		b.Logf("Running http request bench %d/%d", i+1, b.N)
+		cmd := []string{"ab", "-n", fmt.Sprint(times), "-c", fmt.Sprint(*httpConcurrency), serverURL.String()}
+		if stdout, stderr, err := data.runCommandFromPod(testNamespace, benchABPodName, abContainerName, cmd); err != nil {
+			b.Errorf("Error when running http request %dx: %v, stdout: %s, stderr: %s\n", times, err, stdout, stderr)
+		}
+	}
+}
+
+// networkPolicyRealize runs the benchmark of time cost of a network policy with `workloads` amount CIDRs
+// to be realized into flow entries. In order to have entities for the network policy to apply to, we
+// create two dummy Pods: apache-bench and Nginx, they don't have activity during the benchmark test.
+func networkPolicyRealize(workloads int, data *TestData, b *testing.B) {
+	setupPerformanceTestPods(data, b)
+	for i := 0; i < b.N; i++ {
+		go func() {
+			if err := populateWorkload(generateWorkload(workloads), data); err != nil {
+				b.Fatalf("Error when populating workload: %v", err)
+			}
+		}()
+
+		b.StartTimer()
+		if err := waitNetworkPolicyRealize(workloads, data); err != nil {
+			b.Fatalf("Checking network policies realization failed: %v", err)
+		} else {
+			b.Log("Network policy realized")
+		}
+		b.StopTimer()
+
+		if err := data.clientset.NetworkingV1().NetworkPolicies(testNamespace).Delete(workloadNetworkPolicyName, new(metav1.DeleteOptions)); err != nil {
+			b.Fatalf("Error when cleaning up network policies after running one bench iteration: %v", err)
+		}
+	}
+}
+
+func dumpFlows(data *TestData) (string, error) {
+	antreaPodName, err := data.getAntreaPodOnNode(masterNodeName())
+	if err != nil {
+		return "", err
+	}
+	cmd := []string{"ovs-ofctl", "dump-flows", "br-int"}
+	stdout, _, err := data.runCommandFromPod(AntreaNamespace, antreaPodName, "antrea-agent", cmd)
+	if err != nil {
+		return "", err
+	}
+	return stdout, nil
+}
+
+func waitNetworkPolicyRealize(workloads int, data *TestData) error {
```

Comment:
I personally find this much harder to read than the following:
```go
func waitNetworkPolicyRealize(workloads int, data *TestData) error {
    return wait.PollImmediate(time.Second, realizeTimeout, func() (bool, error) {
        return checkRealize(workloads, data)
    })
}
```
Am I missing something? It seems that if you have a timeout value already, using a variant of `Poll` is much more convenient than `WaitFor`. Am I missing something?

Reply:
This is a smarter one. But I would change the interval to `0` to make the function realizes the realization as soon as possible.

Reply:
I'll leave this up-to-you, I don't think there is a big difference here.